### PR TITLE
NN-1795 update prisonstaffhub api

### PR DESCRIPTION
**Pull request by JamieDraperUK**
Upgrade Libraries

Upgrade to Java 11

Remove old auth gateway code

Mirror keyworker-api config

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,24 +1,17 @@
...
+auth.uri.root=${elite2.uri.root}
 jwt.public.key=LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0NCk1JSUJJakFOQmdrcWhraUc5dzBCQVFFRkFBT0NBUThBTUlJQkNnS0NBUUVBc09QQXRzUUFEZGJSdS9FSDZMUDUNCkJNMS9tRjQwVkRCbjEyaEpTWFBQZDVXWUswSExZMjBWTTdBeHhSOW1uWUNGNlNvMVd0N2ZHTnFVeC9XeWVtQnANCklKTnJzLzdEendnM3V3aVF1Tmg0ektSK0VHeFdiTHdpM3l3N2xYUFV6eFV5QzV4dDg4ZS83dk8rbHoxb0NuaXoNCmpoNG14TkFtczZaWUY3cWZuaEpFOVd2V1B3TExrb2prWnUxSmR1c0xhVm93TjdHVEdOcE1FOGR6ZUprYW0wZ3ANCjRveEhRR2hNTjg3SzZqcVgzY0V3TzZEdmhlbWc4d2hzOTZuelFsOG4yTEZ2QUsydXA5UHJyOUdpMkxGZ1R0N0sNCnFYQTA2a0M0S2d3MklSMWVGZ3pjQmxUT0V3bXpqcmU2NUhvTmFKQnI5dU5aelY1c0lMUE1jenpoUWovZk1oejMNCi9RSURBUUFCDQotLS0tLUVORCBQVUJMSUMgS0VZLS0tLS0=
 
-use.api.gateway.auth=false
-api.gateway.token=
-api.gateway.private.key=
-
-flyway.enabled=true
-
 spring.profiles.include=dev
```
Comment (steverendell) [inline src/main/resources/application.properties]: Ref to KW db

File: src/main/java/uk/gov/justice/digital/hmpps/prisonstaffhub/services/HealthInfo.java [hunk 2]
```diff
@@ -21,10 +18,8 @@ public Health health() {
         return Health.up().withDetail("version", getVersion()).build();
     }
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/digital/hmpps/prisonstaffhub/services/HealthInfo.java]: Not keen on losing these javadocs - data served here is wide open and public

File: build.gradle [hunk 3]
```diff
@@ -44,77 +40,92 @@ jar {
...
-    version = 1.18.0
-    sha256 = ""
-}*/
-
 configurations {
     compile.exclude  module: "tomcat-jdbc"
 }
 
 dependencies {
+    annotationProcessor "org.springframework.boot:spring-boot-configuration-processor"
```
Comment (steverendell) [inline build.gradle]: minor - you could go with lombok 1.18.6

File: src/main/java/uk/gov/justice/digital/hmpps/prisonstaffhub/services/HealthInfo.java (hunk 2 above)
Comment (JamieDraperUK) [reply src/main/java/uk/gov/justice/digital/hmpps/prisonstaffhub/services/HealthInfo.java]: I just copied keyworker-api

Comment (JamieDraperUK) [reply src/main/java/uk/gov/justice/digital/hmpps/prisonstaffhub/services/HealthInfo.java]: I just went with whatever keyworker-api does. Not sure why there different.

**Pull request merged**